router/src/server.rs

```diff
@@ -375,11 +499,18 @@ async fn generate_stream(
             metrics::increment_counter!("tgi_request_failure", "err" => "validation");
             tracing::error!("{err}");
             yield Ok(Event::from(err));
-        } else {
+        } else {          
             match infer.generate_stream(req.0).instrument(info_span!(parent: &span, "async_stream")).await {
                 // Keep permit as long as generate_stream lives
                 Ok((_permit, mut response_stream)) => {
                     // Server-Sent Event stream
+                    match stream_type {
+                        StreamType::ChatCompletionsStreamResponse => {
+                            let start_msg = chat_start_message(created_time);
+                            yield Ok(Event::from(Event::default().json_data(start_msg).unwrap()))
```

Comment:
I am not feeling good for `unwrap` call here, although I understand original Hugging Face TGI already includes a ton of `unwrap`.

I wonder if you could handle or convert the error instead of `unwrap` it. 

If `unwrap` is unavoidable, could you add a comment to tell why it is unavoidable.